src/components/ActivityCard/ActivityCard.tsx

```diff
@@ -0,0 +1,274 @@
+import React, { useState } from "react";
+import { Pressable, StyleSheet, Text, View } from "react-native";
+import { Badge, Icon } from "react-native-elements";
+import useCurrentAuthUser from "../../hooks/useCurrentAuthUser";
+import { refetchGetTripQuery } from "../../screens/Itinerary/types/getTripQuery";
+import * as Types from "../../types.d";
+import { useCreateActivityReactionMutation } from "./types/CreateActivityReactionMutation";
+import { useRemoveActivityReactionMutation } from "./types/RemoveActivityReactionMutation";
+
+export interface ActivityCardProps {
+    id: string;
+    name: string;
+    date: Types.Maybe<string> | undefined;
+    activityReactions: Array<{
+        id: string;
+        activityReactionType: Types.ActivityReactionType;
+        addedByUser: { id: string };
+    }>;
+    tripId: string;
+}
+
+export default function ActivityCard(props: ActivityCardProps) {
+    const { getCurrentUser } = useCurrentAuthUser();
+    const [userLiked, setUserLiked] = useState(false);
+    const [userDisliked, setUserDisliked] = useState(false);
+    const [userId, setUserId] = useState("");
+    const [executeCreate] = useCreateActivityReactionMutation();
+    const [executeRemove] = useRemoveActivityReactionMutation();
+
+    const date: Date = new Date(Date.parse(props.date || ""));
+
+    getCurrentUser().then((user) => {
+        if (user == null) {
+            return;
+        }
+        setUserId(user.id);
+
+        setUserLiked(
+            props.activityReactions.filter(
+                (r) =>
+                    r.activityReactionType ===
+                        Types.ActivityReactionType.Like &&
+                    r.addedByUser.id === user.id
+            ).length > 0
+        );
+
+        setUserDisliked(
+            props.activityReactions.filter(
+                (r) =>
+                    r.activityReactionType ===
+                        Types.ActivityReactionType.Dislike &&
+                    r.addedByUser.id === user.id
+            ).length > 0
+        );
+    });
+
+    const handleReactionType = (reactionType: Types.ActivityReactionType) => {
+        const remove = () => {
+            executeRemove({
+                variables: {
+                    input:
+                        props.activityReactions.find(
+                            (r) => r.addedByUser.id === userId
+                        )?.id || "",
```

Comment:
What happens in the latter case ` || ""`? Could we not store this to a variable above `executeRemove()` and only execute it when the reaction id is not `""`? Else this will causes an error in the backend that we can avoid.